lib/SPIRV/libSPIRV/SPIRVValue.h

```diff
@@ -146,69 +146,74 @@ template <spv::Op OC> class SPIRVConstantBase : public SPIRVValue {
   SPIRVConstantBase(SPIRVModule *M, SPIRVType *TheType, SPIRVId TheId,
                     uint64_t TheValue)
       : SPIRVValue(M, 0, OC, TheType, TheId) {
-    Union.UInt64Val = TheValue;
-    recalculateWordCount();
-    validate();
+    setWords(&TheValue);
   }
   // Incomplete constructor for AP integer constant
   SPIRVConstantBase(SPIRVModule *M, SPIRVType *TheType, SPIRVId TheId,
-                    llvm::APInt &TheValue);
+                    const llvm::APInt &TheValue);
   // Complete constructor for float constant
   SPIRVConstantBase(SPIRVModule *M, SPIRVType *TheType, SPIRVId TheId,
                     float TheValue)
       : SPIRVValue(M, 0, OC, TheType, TheId) {
-    Union.FloatVal = TheValue;
-    recalculateWordCount();
-    validate();
+    setWords(reinterpret_cast<uint64_t *>(&TheValue));
```

Comment:
Can we use `getRawData()` here ?
```suggestion
    setWords(TheValue->getRawData());
```